src/gui/guiiconprovider.cpp

```diff
@@ -99,16 +100,15 @@ QIcon GuiIconProvider::generateDifferentSizes(const QIcon &icon)
         return icon;
 
     QIcon newIcon;
-    QList<QSize> requiredSizes;
-    requiredSizes << QSize(16, 16) << QSize(24, 24) << QSize(32, 32);
-    QList<QIcon::Mode> modes;
-    modes << QIcon::Normal << QIcon::Active << QIcon::Selected << QIcon::Disabled;
-    foreach (const QSize &size, requiredSizes) {
-        foreach (QIcon::Mode mode, modes) {
+    const QList<QSize> requiredSizes {{16, 16}, {24, 24}, {32, 32}};
```

Comment:
Please avoid using QList unless it is required. Use QVector instead.

Reply:
Old code, changed.